homeassistant/components/sensor/hue.py

```diff
@@ -0,0 +1,513 @@
+
+    # Hue updates all devices via a single API call.
+    #
+    # If we call a service to update 2 devices, we only want the API to be
+    # called once.
+    #
+    # The throttle decorator will return right away if a call is currently
+    # in progress. This means that if we are updating 2 devices, the first one
+    # is in the update method, the second one will skip it and assume the
+    # update went through and updates it's data, not good!
+    #
+    # The current mechanism will make sure that all devices will wait till
+    # the update call is done before writing their data to the state machine.
+    #
+    # An alternative approach would be to disable automatic polling by Home
+    # Assistant and take control ourselves. This works great for polling as now
+    # we trigger from 1 time update an update to all entities. However it gets
+    # tricky from inside async_turn_on and async_turn_off.
+    #
+    # If automatic polling is enabled, Home Assistant will call the entity
+    # update method after it is done calling all the services. This means that
+    # when we update, we know all commands have been processed. If we trigger
+    # the update from inside async_turn_on, the update will not capture the
+    # changes to the second entity until the next polling update because the
+    # throttle decorator will prevent the call.
+
+    progress = None
+    sensor_progress = set()
+
+    async def request_update(object_id):
+        """Request an update.
+
+        We will only make 1 request to the server for updating at a time. If a
+        request is in progress, we will join the request that is in progress.
+
+        This approach is possible because should_poll=True. That means that
+        Home Assistant will ask sensors for updates during a polling cycle or
+        after it has called a service.
+
+        We keep track of the sensors that are waiting for the request to finish.
+        When new data comes in, we'll trigger an update for all non-waiting
+        sensors. This covers the case where a service is called to modify 2
+        sensors but in the meanwhile some other sensor has changed too.
+        """
+        nonlocal progress
+
+        progress_set = sensor_progress
+        progress_set.add(object_id)
+
+        if progress is not None:
+            return await progress
+
+        progress = asyncio.ensure_future(update_bridge())
+        result = await progress
+        progress = None
+        sensor_progress.clear()
+        return result
+
+    async def update_bridge():
+        """Update the values of the bridge.
+
+        Will update sensors from the bridge.
+        """
+        tasks = []
+        tasks.append(async_update_items(
+            hass, bridge, async_add_devices, request_update,
+            cur_sensors, sensor_progress
+        ))
+
+        await asyncio.wait(tasks)
+
+    await update_bridge()
+
+
+async def async_update_items(hass, bridge, async_add_devices,
```

Comment:
This looks the same as for the binary sensor platform. Try to break out a general function that is imported to the platforms from the hue component.